src/Compilers/CSharp/Test/Semantic/Semantics/LambdaTests.cs

```diff
@@ -8283,6 +8310,32 @@ public void ParamsArray_ParamArrayAttribute()
                 Diagnostic(ErrorCode.ERR_ExplicitParamArray, "System.ParamArray").WithLocation(1, 13));
         }
 
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/66060")]
+        public void ParamsArray_ParamArrayAttribute_OverloadResolution()
+        {
+            var source = """
+                using System;
+
+                new C().M(([ParamArray] int[] xs) => xs.Length);
+                new C().M((params int[] xs) => xs.Length);
+
+                class C
+                {
+                    public void M(D d) { }
+                    public void M(Func<int[], int> f) { }
+                }
+
+                delegate int D(params int[] xs);
+                """;
+            CreateCompilation(source).VerifyDiagnostics(
```

Comment:
A little weird that we're not getting ERR_ExplicitParamArray here. Is it because we never get that far?

Reply:
Yes. That is a declaration diagnostic of the parameter symbol and we never get to report it here since there is the ambiguity error. Without the change in this PR, we would report it (that can be seen in the first commit). But I guess the issue that we are trying to fix here is that we don't want to fail the conversion because the ERR_ExplicitParamArray error is "not fatal", and so the ambiguity error is "more important", that's why this changed.